--- sound/usb/quirks-table.h.orgin  2014-09-01 04:09:59.347304182 +0200
+++ sound/usb/quirks-table.h    2014-09-01 04:09:45.517675720 +0200
@@ -386,6 +386,36 @@ YAMAHA_DEVICE(0x105d, NULL),
    }
 },
 {
+   USB_DEVICE(0x0499, 0x1509),
+   driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
+       /* .vendor_name = "Yamaha", */
+       /* .product_name = "Steinberg UR22", */
+       ifnum = QUIRK_ANY_INTERFACE,
+       type = QUIRK_COMPOSITE,
+       data = (const struct snd_usb_audio_quirk[]) {
+           {
+               ifnum = 1,
+               type = QUIRK_AUDIO_STANDARD_INTERFACE
+           },
+           {
+               ifnum = 2,
+               type = QUIRK_AUDIO_STANDARD_INTERFACE
+           },
+           {
+               ifnum = 3,
+               type = QUIRK_MIDI_YAMAHA
+           },
+           {
+               ifnum = 4,
+               type = QUIRK_IGNORE_INTERFACE
+           },
+           {
+               ifnum = -1
+           }
+       }
+   }
+},
+{
    USB_DEVICE(0x0499, 0x150a),
    .driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
        /* .vendor_name = "Yamaha", */
